src/EnergyPlus/HeatBalanceAirManager.cc

```diff
@@ -4902,6 +4903,25 @@ void ReportZoneMeanAirTemp(EnergyPlusData &state)
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine updates the report variables for the AirHeatBalance.
 
+    if (state.dataHeatBalAirMgr->CalcExtraReportVarMyOneTimeFlag) {
```

Comment:
This is eerily similar to node wet-bulb temp. Shouldn't this be done over in NodeInputManager? and globe temp recorded as a node condition? Then these AirRpt variables could pick up data off the node.

            for (auto const *reqVar : state.dataOutputProcessor->reqVars) {
                if (Util::SameString(reqVar->key, state.dataLoopNodes->NodeID(iNode)) || reqVar->key.empty()) {
                    if (Util::SameString(reqVar->name, "System Node Wetbulb Temperature")) {
                        state.dataNodeInputMgr->NodeWetBulbRepReq(iNode) = true;
                        NodeWetBulbSchedPtr(iNode) = reqVar->SchedPtr;
                    } else if (Util::SameString(reqVar->name, "System Node Relative Humidity")) {
                        NodeRelHumidityRepReq(iNode) = true;
                        NodeRelHumiditySchedPtr(iNode) = reqVar->SchedPtr;
                    } else if (Util::SameString(reqVar->name, "System Node Dewpoint Temperature")) {
                        NodeDewPointRepReq(iNode) = true;
                        NodeDewPointSchedPtr(iNode) = reqVar->SchedPtr;
                    } else if (Util::SameString(reqVar->name, "System Node Specific Heat")) {
                        NodeSpecificHeatRepReq(iNode) = true;
                        NodeSpecificHeatSchedPtr(iNode) = reqVar->SchedPtr;
                    }
                }
            }

Reply:
Is there one node in each space/zone? I didn't find out how to get the node index from space or zone index

Reply:
if this mapping can be done relatively straightforward, then maybe I should calculate it at the nodes above and retrieve them during the reporting.

Reply:
I'm not sure you want to make any big changes here. There is already `Zone/Space Mean Air Temperature` and these changes mimic that report variable. I am just pointing out that there may be a cleaner way to do this. If this PR is clean and gets the job done then that is good enough for now. A new branch could be used to test using the node condition as the basis for these AirRpt variables.

Reply:
I see. I will test it out with some other test branches computing the node condition as the basis for the AirRpt variables.

Reply:
One issue here is that node conditions are updated at the system timestep and these zone values are the averages at the zone timestep. So I'm not sure it's worth pursuing this.

Reply:
Make sense. Maybe we can keep it as this and not calculate at the node level, otherwise, I assume there also needs to be some temporal aggregation steps in addition to node-to-zone/space matching.

Reply:
Yes, keep it like this for now.